mira/metamodel/ops.py

```diff
@@ -116,3 +119,114 @@ def find_models_with_grounding(template_models: Mapping[str, TemplateModel],
     """Filter a dict of models to ones containing a given grounding in any role."""
     return {k: m for k, m in template_models.items()
             if model_has_grounding(m, prefix, identifier)}
+
+
+def simplify_rate_laws(template_model: TemplateModel):
+    """Rewrite templates by simplifying rate laws.
+
+    Parameters
+    ----------
+    template_model :
+        A template model
+
+    Returns
+    -------
+    :
+        A template model with simplified rate laws (overwrites input,
+        not a copy).
+    """
+    new_templates = []
+    for template in template_model.templates:
+        simplified_templates = simplify_rate_law(template,
+                                                 template_model.parameters)
+        if not simplified_templates:
```

Comment:
what is the behavior for empty return list? should this be a more explicit none check?

Reply:
It doesn't return empty lists in any case so `not` here is just checking for `None`, I don't think it make a big difference either way.